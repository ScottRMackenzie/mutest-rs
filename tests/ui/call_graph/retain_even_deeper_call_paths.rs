//@ print-call-graph
//@ print-targets
//@ stdout
//@ stderr: empty
//@ mutest-flags: --depth=6

fn non_recursive_function_a() {}

fn non_recursive_function_b() {
    non_recursive_function_a();
}

fn non_recursive_function_c() {
    non_recursive_function_b();
}

#[test]
fn test() {
    non_recursive_function_c();
}

#[test]
fn other_test() {
    non_recursive_function_a();
}
