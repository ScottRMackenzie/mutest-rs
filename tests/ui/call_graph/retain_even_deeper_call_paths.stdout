
@@@ call graph @@@

entry points:

test other_test
  -> non_recursive_function_a at tests/ui/call_graph/retain_even_deeper_call_paths.rs:7:1: 7:30 (#0)
test test
  -> non_recursive_function_c at tests/ui/call_graph/retain_even_deeper_call_paths.rs:13:1: 13:30 (#0)

nested calls at distance 1:

non_recursive_function_c at tests/ui/call_graph/retain_even_deeper_call_paths.rs:13:1: 13:30 (#0)
  -> non_recursive_function_b at tests/ui/call_graph/retain_even_deeper_call_paths.rs:9:1: 9:30 (#0)

nested calls at distance 2:

non_recursive_function_b at tests/ui/call_graph/retain_even_deeper_call_paths.rs:9:1: 9:30 (#0)
  -> non_recursive_function_a at tests/ui/call_graph/retain_even_deeper_call_paths.rs:7:1: 7:30 (#0)

nested calls at distance 3:


nested calls at distance 4:


nested calls at distance 5:


@@@ targets @@@

tests -(0)-> non_recursive_function_a at tests/ui/call_graph/retain_even_deeper_call_paths.rs:7:1: 7:30 (#0)
  (0) other_test
  (2) test

tests -(1)-> non_recursive_function_b at tests/ui/call_graph/retain_even_deeper_call_paths.rs:9:1: 9:30 (#0)
  (1) test

tests -(0)-> non_recursive_function_c at tests/ui/call_graph/retain_even_deeper_call_paths.rs:13:1: 13:30 (#0)
  (0) test

targets: 3 total; 3 safe; 0 unsafe (0 tainted)
